src/reconcile/rec_visibility.c

```diff
@@ -290,6 +292,16 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
     /* Keep track of the selected update. */
     upd = upd_select->upd;
 
+#ifdef HAVE_DIAGNOSTIC
+    /* Double check the update is visible if running with read uncommitted. */
+    if (!is_hs_page && !WT_IS_METADATA(session->dhandle) &&
```

Comment:
Not sure why I have to add `!WT_IS_METADATA(session->dhandle)`? Does metadata have special visiblity rules?